src/components/organisms/WalletList/walletList.organism.tsx

```diff
@@ -0,0 +1,87 @@
+import React, {useEffect} from "react";
+import { useDispatch } from "react-redux";
+import {WalletConnectConnector} from "@web3-react/walletconnect-connector";
+import Typography from "../../atoms/Typography/typography.atom";
+import {
+  CONNECTOR_LABELS,
+  CONNECTOR_LOGOS,
+  CONNECTOR_DESCRIPTIONS,
+  supportedConnectorList,
+} from "../../../shared/constants/web3.constants";
+import { EColor, EFontWeight } from "../../../shared/types/styles.types";
+import {EConnectorType} from "../../../shared/types/web3.types";
+import styles from "./walletList.organism.module.scss";
+import WalletService from "../../../shared/services/wallet/wallet.service";
+import useWeb3 from "../../../hooks/useWeb3";
+import {EModalName} from "../../../redux/ui/ui.redux.types";
+import {toggleModal} from "../../../redux/ui/ui.redux.actions";
+import {EErrorTypes} from "../../../shared/types/error.types";
+
+const WalletList: React.FC = () => {
+  const dispatch = useDispatch();
+  const { activate, active, error, connector, mappedError } = useWeb3();
+
+  const closeModal = () => dispatch(toggleModal(EModalName.CONNECT_WALLET_V2, false));
+
+  const connectWalletFnFactory = (type_: EConnectorType) => {
+    return () => {
+      const _connector = WalletService.typeToProvider(type_);
+      activate(_connector).finally(() => closeModal());
+    };
+  };
+
+  useEffect(() => {
+    if (error) {
+      if (connector instanceof WalletConnectConnector) {
+        if (mappedError === EErrorTypes.UNSUPPORTED_CHAIN) {
+          closeModal();
```

Comment:
Hmmm, not sure if we should close the modal here, maybe the user wants to switch to metamask instead

Reply:
This is how it behaves in the previous modal.